include/envoy/network/connection.h

```diff
@@ -120,14 +120,15 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return The address of the remote client.
    */
   virtual const Address::Instance& remoteAddress() PURE;
-
+  virtual Address::InstanceConstSharedPtr remoteAddressConstSharedPtr() PURE;
   /**
    * @return the local address of the connection. For client connections, this is the origin
    * address. For server connections, this is the local destination address. For server connections
    * it can be different from the proxy address if the downstream connection has been redirected or
    * the proxy is operating in transparent mode.
    */
   virtual const Address::Instance& localAddress() PURE;
+  virtual Address::InstanceConstSharedPtr localAddressConstSharedPtr() PURE;
```

Comment:
The issue is https://github.com/lyft/envoy/blob/master/source/common/network/cidr_range.cc#L88. If you go back to the original https://github.com/lyft/envoy/pull/1090/commits/64e5287b8504224edaf247547eba45648af2e657, this is what @hennna did. But, the implementation was via string conversion and CIDR range construction.

It strikes me (as a bit twiddler), that we should be able to perform this operation way faster than string parsing and object construction. It should just be two right shift operations and equality. This speaks to the underlying `Address::Instance` representation at EOD.